code/datums/components/deadchat_control.dm

```diff
@@ -0,0 +1,134 @@
+#define DEMOCRACY_MODE "democracy"
+#define ANARCHY_MODE "anarchy"
+
+/datum/component/deadchat_control
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	var/list/inputs = list("up", "down", "left", "right", "north", "south", "west", "east")
+	var/list/ckey_to_cooldown = list()
+	var/deadchat_mode = DEMOCRACY_MODE
+	var/democracy_cooldown
+	var/anarchy_cooldown
+	var/force_move
+	var/require_orbiting
+
+/datum/component/deadchat_control/Initialize(_deadchat_mode, _democracy_cooldown = 12 SECONDS, _anarchy_cooldown = 20 SECONDS, _force_move = TRUE, _require_orbiting = FALSE)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(SSdcs, COMSIG_GLOB_DEAD_SAY, .proc/deadchat_react)
+	deadchat_mode = _deadchat_mode
+	if(deadchat_mode == DEMOCRACY_MODE)
+		addtimer(CALLBACK(src, .proc/democracy_loop), democracy_cooldown)
+	democracy_cooldown = _democracy_cooldown
+	anarchy_cooldown = _anarchy_cooldown
+	force_move = _force_move
+	require_orbiting = _require_orbiting
+
+/datum/component/deadchat_control/proc/deadchat_react(datum/source, mob/player, message)
+	var/atom/movable/A = parent
+	if(require_orbiting)
+		if(!(player in A.orbiters?.orbiters))
+			return
+	if(deadchat_mode == ANARCHY_MODE)
+		var/cooldown = ckey_to_cooldown[player.ckey]
+		if(cooldown)
+			return
+		var/direction
+		message = lowertext(message)
+		switch(message)
```

Comment:
A better way to do this would be to have a list of `list(command = callback)`

If you need help with that or any other component thing feel free to contact me